pkgs/http_client_conformance_tests/lib/src/request_body_tests.dart

```diff
@@ -35,189 +35,121 @@ class _Plus2Encoding extends Encoding {
-          unawaited(request.response.close());
-        });
-      await client.post(Uri.http('localhost:${server.port}', ''),
-          headers: {HttpHeaders.contentTypeHeader: 'image/png'},
-          body: [1, 2, 3, 4, 5]);
+      final serverReceivedContentType = await httpServerQueue.next;
+      final serverReceivedBody = await httpServerQueue.next as String;
 
       expect(serverReceivedContentType, ['image/png']);
       expect(serverReceivedBody.codeUnits, [1, 2, 3, 4, 5]);
-      await server.close();
     });
 
     test('client.post() with List<int> with encoding', () async {
       // Encoding should not affect binary payloads.
-      late String serverReceivedBody;
-
-      final server = (await HttpServer.bind('localhost', 0))
-        ..listen((request) async {
-          serverReceivedBody = await const Utf8Decoder()
-              .bind(request)
-              .fold('', (p, e) => '$p$e');
-          unawaited(request.response.close());
-        });
-      await client.post(Uri.http('localhost:${server.port}', ''),
+      await client.post(Uri.http(host, ''),
           body: [1, 2, 3, 4, 5], encoding: _Plus2Encoding());
 
+      await httpServerQueue.next; // Content-Type.
+      final serverReceivedBody = await httpServerQueue.next as String;
+
       // RFC 2616 7.2.1 says that:
       //   Any HTTP/1.1 message containing an entity-body SHOULD include a
       //   Content-Type header field defining the media type of that body.
       // But we didn't set one explicitly so don't verify what the server
       // received.
       expect(serverReceivedBody.codeUnits, [1, 2, 3, 4, 5]);
-      await server.close();
     });
 
     test('client.post() with List<int> with encoding and content-type',
         () async {
       // Encoding should not affect the payload but it should affect the
       // content-type.
-      late List<String>? serverReceivedContentType;
-      late String serverReceivedBody;
 
-      final server = (await HttpServer.bind('localhost', 0))
-        ..listen((request) async {
-          serverReceivedContentType =
-              request.headers[HttpHeaders.contentTypeHeader];
-          serverReceivedBody = await const Utf8Decoder()
-              .bind(request)
-              .fold('', (p, e) => '$p$e');
-          unawaited(request.response.close());
-        });
-      await client.post(Uri.http('localhost:${server.port}', ''),
-          headers: {HttpHeaders.contentTypeHeader: 'image/png'},
+      await client.post(Uri.http(host, ''),
+          headers: {'Content-Type': 'image/png'},
           body: [1, 2, 3, 4, 5],
           encoding: _Plus2Encoding());
 
+      final serverReceivedContentType = await httpServerQueue.next;
+      final serverReceivedBody = await httpServerQueue.next as String;
+
       expect(serverReceivedContentType, ['image/png; charset=plus2']);
       expect(serverReceivedBody.codeUnits, [1, 2, 3, 4, 5]);
-      await server.close();
     });
-
+/*
```

Comment:
Delete this instead of commenting it out. We can find it in source history if we need it again.

Reply:
I didn't mean to leave that - it was just sloppiness of my part.